### PR TITLE
Adding notes about providing code changes and pull requests.

### PR DESCRIPTION
**Pull request by destogl**

### CODE REVIEW

File: docs/guidelines/development.rst [hunk 1]
```diff
@@ -7,15 +7,68 @@ This documents proposes development guidelines to increase work efficiency and s
...
-===============================================
+1. `Making changes to the code-base by submitting a Pull Request <#>`_
+
+2. `Using *pre-commit* <#use-pre-commit-for-formatting-and-linting>`_
+
+
+Code Changes and Pull Request Submissions
+==========================================
+
+To make a change of the existing code base Pull Request (GitHub) or Merge Request (GitLab) are used.
```
Comment (Robotgir) [inline docs/guidelines/development.rst:21]: To make a change of the existing code base, Pull Request (GitHub) or Merge Request (GitLab) are used.

ADDING COMMA WILL BE MORE CLEAR

Comment (Robotgir) [inline docs/guidelines/development.rst:21]: To make a change of the existing code base, Pull Request (GitHub) or Merge Request (GitLab) are used.

File: docs/guidelines/development.rst [hunk 2]
```diff
@@ -7,15 +7,68 @@ This documents proposes development guidelines to increase work efficiency and s
...
+This section describe in short process with valuable tips to make your and reviewers' life easier.
```
Comment (Robotgir) [inline docs/guidelines/development.rst:22]: This section describes in

File: docs/guidelines/development.rst [hunk 3]
```diff
@@ -7,15 +7,68 @@ This documents proposes development guidelines to increase work efficiency and s
...
+#. **Always** create a new branch for each feature or bug fix. Don't submit PRs from *master* branch.
+
+#. **Before** starting development check how the branch will be merged, using *merge commit* or *squash* method.
+   If *merge commit* is used make sure that each of your commits is clean and named properly since they will become part of the repository's history.
+
+   .. note::
+
+      In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.
+
+#. Explain in the PR/MR description what you code is doing and why.
```
Comment (Robotgir) [inline docs/guidelines/development.rst:47]: Explain in the PR/MR description what your

File: docs/guidelines/development.rst [hunk 4]
```diff
@@ -7,15 +7,68 @@ This documents proposes development guidelines to increase work efficiency and s
...
+
+#. When you are finished with development and want to submit code for the review - consider the following tips:
+
+   - **Always** run *pre-commit* formaters;
+   - Review your code **first by yourself** before asking someone else;
+   - Make sure there **are not** commended code blocks or if they have to be there add explanation why;
```
Comment (Robotgir) [inline docs/guidelines/development.rst:53]: - Make sure there **are not** commented code

File: docs/guidelines/development.rst (hunk 3 above)
Comment (destogl) [reply docs/guidelines/development.rst:47]:
```suggestion
#. EExplain in the PR/MR description what your code is doing and why.
```

File: docs/guidelines/development.rst (hunk 2 above)
Comment (destogl) [reply docs/guidelines/development.rst:22]:
```suggestion
This section describes in short process with valuable tips to make your and reviewers' life easier.
```

File: docs/guidelines/development.rst (hunk 1 above)
Comment (destogl) [reply docs/guidelines/development.rst:21]:
```suggestion
To make a change of the existing code base, Pull Request (GitHub) or Merge Request (GitLab) are used.
```

File: docs/guidelines/development.rst [hunk 5]
```diff
@@ -7,15 +7,68 @@ This documents proposes development guidelines to increase work efficiency and s
...
+   Therefore be careful about that and use the opportunity to sync *master* branch of the fork to the state of the *master* branch of the upstream repository.
+
+#. **Always** create a new branch for each feature or bug fix. Don't submit PRs from *master* branch.
+
+#. **Before** starting development check how the branch will be merged, using *merge commit* or *squash* method.
+   If *merge commit* is used make sure that each of your commits is clean and named properly since they will become part of the repository's history.
+
+   .. note::
+
+      In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.
```
Comment (Robotgir) [inline docs/guidelines/development.rst:45]: `In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.`

Comment (Robotgir) [inline docs/guidelines/development.rst:45]: `In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.`

Comment (Robotgir) [inline docs/guidelines/development.rst:45]: `In *Stogl Robotics* we are always *squashing* commits, i.e., one feature or one bug fix is one commit in the default branch. The commit message is edited before merging.`

**Pull request merged**